### PR TITLE
Refactor PlayButton

### PR DESCRIPTION
**Pull request by paulogama14**
# Goal

- Refactor `PlayButton.swift`

### CODE REVIEW

File: Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
-        return .center
-    }
+    override open var panel: MediaControlPanel { .center}
+    override open var position: MediaControlPosition { .center }
 
     public var playIcon = UIImage.fromName("play", for: PlayButton.self)!
     public var pauseIcon = UIImage.fromName("pause", for: PlayButton.self)!
 
-    public var button: UIButton! {
+    public var button: UIButton? {
```
Comment (helbertgs) [inline Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift:10]: I think this way can be a good idea. 

```suggestion
    public var button: UIButton = {
        let view = UIButton(type: .custom)
        view.accessibilityIdentifier = "PlayPauseButton"
        view.setImage(playIcon, for: .normal)
        view.imageView?.contentMode = .scaleAspectFit
        
        return view
    }()
```

File: Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift [hunk 2]
```diff
@@ -71,26 +81,35 @@ open class PlayButton: MediaControl.Element {
...
+    
+    private func changeToPlayIcon() {
+        guard canShowPlayIcon else { return }
 
-        if playback.state == .paused || playback.state == .idle {
-            button.setImage(playIcon, for: .normal)
-        } else if playback.state == .playing {
-            button.setImage(pauseIcon, for: .normal)
-        }
+        button?.setImage(playIcon, for: .normal)
```
Comment (helbertgs) [inline Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift:107]:
```suggestion
        button.setImage(playIcon, for: .normal)
```

File: Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift [hunk 3]
```diff
@@ -71,26 +81,35 @@ open class PlayButton: MediaControl.Element {
...
+    }
+
+    public func changeToPauseIcon() {
+        guard canShowPauseIcon else { return }
+
+        button?.setImage(pauseIcon, for: .normal)
```
Comment (helbertgs) [inline Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift:113]:
```suggestion
        button.setImage(pauseIcon, for: .normal)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 4]
```diff
@@ -71,13 +71,13 @@ class PlayButtonTests: QuickSpec {
                     }
 
                     it("calls the playback play") {
-                        playButton.button.sendActions(for: .touchUpInside)
+                        playButton.button?.sendActions(for: .touchUpInside)
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:74]:
```suggestion
                        playButton.button.sendActions(for: .touchUpInside)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 5]
```diff
@@ -71,13 +71,13 @@ class PlayButtonTests: QuickSpec {
...
 
                         expect(coreStub.playbackMock?.didCallPlay).to(beTrue())
                     }
 
                     it("shows play button") {
-                        playButton.button.sendActions(for: .touchUpInside)
+                        playButton.button?.sendActions(for: .touchUpInside)
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:80]:
```suggestion
                        playButton.button.sendActions(for: .touchUpInside)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 6]
```diff
@@ -89,13 +89,13 @@ class PlayButtonTests: QuickSpec {
                     }
 
                     it("calls the playback play") {
-                        playButton.button.sendActions(for: .touchUpInside)
+                        playButton.button?.sendActions(for: .touchUpInside)
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:92]:
```suggestion
                        playButton.button.sendActions(for: .touchUpInside)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 7]
```diff
@@ -114,17 +114,17 @@ class PlayButtonTests: QuickSpec {
                     }
 
                     it("calls the playback pause") {
-                        playButton.button.sendActions(for: .touchUpInside)
+                        playButton.button?.sendActions(for: .touchUpInside)
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:117]:
```suggestion
                        playButton.button.sendActions(for: .touchUpInside)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 8]
```diff
@@ -114,17 +114,17 @@ class PlayButtonTests: QuickSpec {
...
 
                         expect(coreStub.playbackMock?.didCallPause).to(beTrue())
                     }
 
                     it("changes the image to a play icon") {
                         let playIcon = UIImage.fromName("play", for: PlayButton.self)!
 
-                        playButton.button.sendActions(for: .touchUpInside)
+                        playButton.button?.sendActions(for: .touchUpInside)
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:125]:
```suggestion
                        playButton.button.sendActions(for: .touchUpInside)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 9]
```diff
@@ -114,17 +114,17 @@ class PlayButtonTests: QuickSpec {
...
 
-                        let currentButtonIcon = (playButton.button.imageView?.image)!
+                        let currentButtonIcon = (playButton.button?.imageView?.image)!
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:127]:
```suggestion
                        let currentButtonIcon = (playButton.button.imageView?.image)!
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 10]
```diff
@@ -148,17 +148,17 @@ class PlayButtonTests: QuickSpec {
                     }
 
                     it("calls the playback play") {
-                        playButton.button.sendActions(for: .touchUpInside)
+                        playButton.button?.sendActions(for: .touchUpInside)
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:151]:
```suggestion
                        playButton.button.sendActions(for: .touchUpInside)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 11]
```diff
@@ -148,17 +148,17 @@ class PlayButtonTests: QuickSpec {
...
 
                         expect(coreStub.playbackMock?.didCallPlay).to(beTrue())
                     }
 
                     it("changes the image to a pause icon") {
                         let pauseIcon = UIImage.fromName("pause", for: PlayButton.self)!
 
-                        playButton.button.sendActions(for: .touchUpInside)
+                        playButton.button?.sendActions(for: .touchUpInside)
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:159]:
```suggestion
                        playButton.button.sendActions(for: .touchUpInside)
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 12]
```diff
@@ -148,17 +148,17 @@ class PlayButtonTests: QuickSpec {
...
 
-                        let currentButtonIcon = (playButton.button.imageView?.image)!
+                        let currentButtonIcon = (playButton.button?.imageView?.image)!
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:161]:
```suggestion
                        let currentButtonIcon = (playButton.button.imageView?.image)!
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 13]
```diff
@@ -181,7 +181,7 @@ class PlayButtonTests: QuickSpec {
                 it("set's acessibilityIdentifier to button") {
                     playButton.render()
 
-                    expect(playButton.button.accessibilityIdentifier).to(equal("PlayPauseButton"))
+                    expect(playButton.button?.accessibilityIdentifier).to(equal("PlayPauseButton"))
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:184]:
```suggestion
                    expect(playButton.button.accessibilityIdentifier).to(equal("PlayPauseButton"))
```

File: Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift [hunk 14]
```diff
@@ -194,7 +194,7 @@ class PlayButtonTests: QuickSpec {
                     it("has scaleAspectFit content mode") {
                         playButton.render()
 
-                        expect(playButton.button.imageView?.contentMode).to(equal(UIView.ContentMode.scaleAspectFit))
+                        expect(playButton.button?.imageView?.contentMode).to(equal(UIView.ContentMode.scaleAspectFit))
```
Comment (helbertgs) [inline Tests/Clappr_Tests/Classes/Plugin/Core/MediaControl/PlayButtonTests.swift:197]:
```suggestion
                        expect(playButton.button.imageView?.contentMode).to(equal(UIView.ContentMode.scaleAspectFit))
```

File: Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift [hunk 15]
```diff
@@ -24,42 +19,57 @@ open class PlayButton: MediaControl.Element {
...
+    }
 
+    private func setupView() {
+        guard let superview = view.superview else { return }
+
+        view.widthAnchor.constraint(equalTo: superview.widthAnchor, multiplier: 0.6).isActive = true
+        view.heightAnchor.constraint(equalTo: superview.heightAnchor, multiplier: 0.6).isActive = true
+    }
+
+    private func setupButton() {
```
Comment (helbertgs) [inline Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift:55]: How you wont to make constraints, you can remove this function

File: Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift (hunk 1 above)
Comment (paulogama14) [reply Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift:10]: Thank you. We have to assign the button inside the view from `MediaControl.Element` and we create the button instance on `Element.render()` to conform with the architecture. 
What do you think?

Comment (helbertgs) [reply Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift:10]: OK! LGTM

**Pull request merged**